### PR TITLE
Extrapolation update

### PR DESCRIPTION
**Pull request by raphaeldussin**
refreshed the old PR. Needs more documentation but works great!

### CODE REVIEW

**Comment by raphaeldussin** [comment]
@RondeauG @trondkr this is an update from PR https://github.com/JiaweiZhuang/xESMF/pull/97
Let me know if you have comments. Note that I also have a PR to change masked values from zero to NaN here:
https://github.com/pangeo-data/xESMF/pull/19

**Review by huard — COMMENTED** [review]
I suggest xESMF sticks as much as possible to the same variable names used in ESMF and ESMPy.

File: xesmf/frontend.py [hunk 1]
```diff
@@ -158,6 +159,20 @@ def __init__(self, ds_in, ds_out, method, periodic=False,
...
+            - 'nearest_s2d'
+
+        extrap_exp : float, optional
+            The exponent to raise the distance to when calculating weights for the
+            extrapolation method. If none are specified, defaults to 2.0
+
+        extrap_num_pnts : int, optional
+            The number of source points to use for the extrapolation methods
+            that use more than one source point. If none are specified, defaults to 8
+
```
[the remark this answers is not in the document]
Comment (raphaeldussin) [reply xesmf/frontend.py:175]: I agree! and this is a pretty minor change :)

**Comment by raphaeldussin** [comment]
@huard thanks for the review. I should be able to wrap this up pretty sson.

related to this, I noticed ESMPy does not provide the API for the CREEP_FILL extrapolation method so I submitted a PR
https://github.com/esmf-org/esmf/pull/18
and will put an issue in here to remind us to add it to xESMF once this become available

**Comment by RondeauG** [comment]
> @RondeauG @trondkr this is an update from PR [JiaweiZhuang#97](https://github.com/JiaweiZhuang/xESMF/pull/97)
> Let me know if you have comments. Note that I also have a PR to change masked values from zero to NaN here:
> #19

Hi @raphaeldussin, I'm happy to see that this option will soon be integrated. I looked at your changes and everything looks fine. Great work!

**Review by huard — APPROVED** [review]

**Comment by raphaeldussin** [comment]
I don't understand why the RTD build failed. Should I merge anyway and we can try to figure out what happened and fix it in another PR?

**Comment by huard** [comment]
Me neither, yes, go ahead.

**Pull request merged**